### PR TITLE
feat(datasource): use the relocation tag to define "replacement" changes for maven.

### PR DESCRIPTION
**Pull request by astellingwerf**
## Changes:

Use the [Maven `relocation` tag](https://maven.apache.org/guides/mini/guide-relocation.html) to set the replacement name and version (as added by #5558) to automatically create replacements.

## Context:

Add to the functionality of #5558, fixing several related issues.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/process/lookup/index.ts [hunk 1]
```diff
@@ -74,6 +74,13 @@ export async function lookupUpdates(
     if (dependency.deprecationMessage) {
       logger.debug({ dependency: depName }, 'Found deprecationMessage');
       res.deprecationMessage = dependency.deprecationMessage;
+      if (dependency.replacementName && dependency.replacementVersion) {
```
Comment (astellingwerf) [inline lib/workers/repository/process/lookup/index.ts:77]: This looks to be a mismerge.

**Comment by HonkingGoose** [comment]
There seems to be a problem with the commits on this PR, there's a commit from jamiemagee here, that I don't think should be there?

Might be good to create a new PR with a clean history?

**Comment by astellingwerf** [comment]
The way that jamiemagee has created and recreated the branche for #5558 is the reason why I couldn't seem to do it right. To get (the June edition of ) his branch in my fork, I had to merge and resolve conflicts. Any tips on how I keep branches (other than master) in my fork in sync?

**Comment by JamieMagee** [comment]
I've taken to squashing that branch and force pushing it to make it simpler to keep up-to-date with `main`.

Just to clarify, this adds the replacement functionality for maven, that I added for npm in #5558?

**Comment by astellingwerf** [comment]
Are you saying your change is exclusive to NPM?

Yes, my change reads the replacement (relocation in Maven lingo) data from POM files in Maven repos. With that, it can generate the replacements, without hardcoding it in Renovate itself. This would resolve issues where Renovate did upgrade packages to versions that only have the relocation metadata, and no actual artifacts.

**Comment by astellingwerf** [comment]
I just figured out how to run Renovate locally (in the IDE), so I'm able to do more e2e debugging. Let me have a look/do you know why for replacements there's a `{package-name}-.x` branch?

**Comment by JamieMagee** [comment]
It will work with any manager, but that PR only implements it for npm.

I'm not sure about the branch naming, but I think your guess is likely correct.

**Comment by astellingwerf** [comment]
Could it be that this doesn't work for Gradle, because you implemented the change for the NPM updater, but not the more generic [`lib/workers/branch/get-updated.ts`](lib/workers/branch/get-updated.ts), let alone the `updateDependency` implementation of the different managers?

**Pull request closed without merging**